src/storage/src/table/cell_based_table.rs

```diff
@@ -273,13 +273,14 @@ impl<S: StateStore, SER: RowSerializer, const T: AccessType> CellBasedTableBase<
         tracing::trace!(target: "events::storage::cell_based_table", "compute vnode: {:?} keys {:?} => {}", row, indices, vnode);
 
         // This table should only be used to access entries with vnode specified in `self.vnodes`.
-        assert!(
-            self.vnodes.is_set(vnode as usize).unwrap(),
-            "vnode {} should not be accessed by this table: {:#?}, dist key {:?}",
-            vnode,
-            self.table_columns,
-            self.dist_key_indices
-        );
+        // FIXME: use assert when batch scan pruning is implemented.
+        // assert!(
+        //     self.vnodes.is_set(vnode as usize).unwrap(),
+        //     "vnode {} should not be accessed by this table: {:#?}, dist key {:?}",
+        //     vnode,
+        //     self.table_columns,
+        //     self.dist_key_indices
+        // );
```

Comment:
I mean https://github.com/singularity-data/risingwave/issues/3450 is not implemented, i.e., the executor will still be created even if its partition doesn't need to be read

Reply:
But I can let the executor early exits now 🤔🥸

Reply:
Do you mean point getting will still schedule multiple seq scan executors? I think the scheduler should only create one executor and assign full vnodes for it, instead of hacking on the cell-based table implementation.

Reply:
Removing this assertion will seriously affect the debugging of distribution keys like #3573 #3572...

Reply:
I was planning to do it in a next PR. But I can workaround it in executor now.